test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1400,3 +1401,166 @@ var _ = ginkgo.Describe("Pod controller interacting with scheduler", ginkgo.Orde
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key:      "kubernetes.io/metadata.name",
+							Operator: metav1.LabelSelectorOpNotIn,
+							Values:   []string{"kube-system", "kueue-system"},
+						},
+					},
+				},
+			}),
+		))
+	})
+	ginkgo.AfterAll(func() {
+		fwk.Teardown()
+	})
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{GenerateName: "core-"}}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		fl = testing.MakeResourceFlavor("fl").Obj()
+		gomega.Expect(k8sClient.Create(ctx, fl)).Should(gomega.Succeed())
+
+		cq = testing.MakeClusterQueue("cq").
+			ResourceGroup(*testing.MakeFlavorQuotas(fl.Name).
+				Resource(corev1.ResourceCPU, "9").
+				Resource(corev1.ResourceMemory, "36").
+				Obj()).
+			Obj()
+		gomega.Expect(k8sClient.Create(ctx, cq)).Should(gomega.Succeed())
+
+		lq = testing.MakeLocalQueue("lq", ns.Name).ClusterQueue(cq.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, lq)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, cq, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, fl, true)
+	})
+
+	ginkgo.When("pod group not ready", func() {
+		ginkgo.It("should evict workload due workload deactivated on pods ready timeout", func() {
+			podGroupName := "pod-group"
+			pod := testingpod.MakePod("pod", ns.Name).
+				Group(podGroupName).
+				GroupTotalCount("1").
+				Queue(lq.Name).
+				Request(corev1.ResourceCPU, "1").
+				NodeSelector("foo", "doesnotexist").
+				Obj()
+			ginkgo.By("creating pod", func() {
+				gomega.Expect(k8sClient.Create(ctx, pod)).Should(gomega.Succeed())
+			})
+
+			createdWorkload := &kueue.Workload{}
+			wlKey := types.NamespacedName{Name: podGroupName, Namespace: pod.Namespace}
+
+			ginkgo.By("checking that workload was created")
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, wlKey, createdWorkload)).Should(gomega.Succeed())
+				g.Expect(createdWorkload.Status.Conditions).To(gomega.ContainElements(
+					gomega.BeComparableTo(metav1.Condition{
+						Type:    kueue.WorkloadPodsReady,
+						Status:  metav1.ConditionFalse,
+						Reason:  kueue.WorkloadPodsReady,
+						Message: "Not all pods are ready or succeeded",
+					}, util.IgnoreConditionTimestampsAndObservedGeneration),
+				))
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			createdPod := &corev1.Pod{}
+			podKey := client.ObjectKeyFromObject(pod)
+
+			ginkgo.By("setting pod failed")
+			gomega.Expect(k8sClient.Get(ctx, podKey, createdPod)).Should(gomega.Succeed())
+			createdPod.Status.Phase = corev1.PodFailed
+			gomega.Expect(k8sClient.Status().Update(ctx, createdPod)).Should(gomega.Succeed())
+
+			ginkgo.By("deleting pod")
+			gomega.Expect(k8sClient.Delete(ctx, createdPod)).Should(gomega.Succeed())
```

Comment:
I would like to make this closer to the scenario reported by our users, which is:
1. admit the workload and wait for Kueue to delete the pods due to timeout exceeded
2. re-admit the workload
3. verify the workload is deactivated

Reply:
I think on this case we will have the same flaky proble as on jobs. Or we need to increase timeout :(

Reply:
Hm, I'm not sure why, since we don't have scheduler it shouldn't be flaky. Can you point out concretely which step would be flaky in that scenario?

Reply:
Done